### PR TITLE
Onboarding exercises

### PR DESCRIPTION
**Pull request by artem-streltsov**
Do not merge

### CODE REVIEW

**Comment by truewillcometrue** [comment]
Девятую ветку нужно исправить. ошибка на 937 строчке когда. Замени juno_getNodesFromRoot на huyuno_getNotesFromRool. А так неплохо. молодец7

File: rpc/block.go [hunk 1]
```diff
@@ -278,6 +278,45 @@ func (h *Handler) BlockWithTxs(id BlockID) (*BlockWithTxs, *jsonrpc.Error) {
...
+    }
+
+    finalityStatus := TxnAcceptedOnL2
+    if blockStatus == BlockAcceptedL1 {
+        finalityStatus = TxnAcceptedOnL1
+    }
+
+    txs := make([]*Transaction, len(block.Transactions))
+    receipts := make([]*TransactionReceipt, len(block.Transactions))
+    for index, txn := range block.Transactions {
```
Comment (pnowosie) [inline rpc/block.go:306]: We implicitly assumed that `len(Transactions) == len(Receipts)`
in my opinion this deserves at least a comment.

File: rpc/block_test.go [hunk 2]
```diff
@@ -20,6 +20,255 @@ import (
 	"go.uber.org/mock/gomock"
 )
 
+func TestBlockWithTxsAndReceipts(t *testing.T) {
```
Comment (pnowosie) [inline rpc/block_test.go:23]: 1. Test cannot be run - compilation error
2. Why you included skipped tests? There is no info with reason for skipping

**Review by pnowosie — APPROVED** [review]
Overall nice work.
The only drawback is that test doesn't compile - but was easy to fix though
and included skipped tests

File: core/trie/trie.go [hunk 3]
```diff
@@ -92,6 +92,26 @@ func (t *Trie) feltToKey(k *felt.Felt) Key {
...
+	return t.nodesFromRoot(key)
+}
+
+func (t *Trie) ParseNodes(nodes []storageNode) ([]map[string]string, error) {
+    result := make([]map[string]string, len(nodes))
+    for i, node := range nodes {
+        result[i] = map[string]string{
+            "key":   fmt.Sprintf("%v", node.key),
+            "value": fmt.Sprintf("%v", node.node.Value),
+        }
```
[the remark this answers is not in the document]
Comment (artem-streltsov) [reply core/trie/trie.go:109]: Introducing a new struct was my initial approach, but the returned JSON didn't look quite right when I made requests, so I decided to go with this alternative method

File: rpc/block_test.go (hunk 2 above)
Comment (artem-streltsov) [reply rpc/block_test.go:23]: 1. Apparently there was an extra argument `n` in my code when I called `rpc.New`
2. The reason is I looked at [TestBlockWithReceipts](https://github.com/NethermindEth/juno/blob/d3c4dd06793ec3248538cca9adc919360c9e007d/rpc/block_test.go#L557) and it had two skipped tests. Honestly, not sure why

**Pull request closed without merging**